trino/auth.py

```diff
@@ -142,101 +143,77 @@ class _OAuth2TokenBearer(AuthBase):
-            return r
+    def __authenticate(self, response, **kwargs):
+        if not 400 <= response.status_code < 500:
+            return response
 
         # we have to handle the authentication, may be token the token expired or it wasn't there at all
-        auth_info = r.headers.get('WWW-Authenticate')
+        auth_info = response.headers.get('WWW-Authenticate')
         if not auth_info:
             raise exceptions.TrinoAuthError("Error: header WWW-Authenticate not available in the response.")
 
-        matches = dict((s.strip().split('=', 1) for s in auth_info.split(",")))
-        if matches is None:
+        matches = self._pattern.match(auth_info)
+        if not matches:
             raise exceptions.TrinoAuthError(f"Error: header info didn't match {auth_info}")
 
-        auth_server = matches.get('Bearer x_redirect_server')
-        if auth_server is None:
-            raise exceptions.TrinoAuthError("Error: header info didn't have Bearer x_redirect_server")
-
-        token_server = matches.get('x_token_server')
-        if token_server is None:
-            raise exceptions.TrinoAuthError("Error: header info didn't have x_token_server")
-
-        self._thread_local.token_server = token_server.strip('"')
+        redirect_server = matches.group(1)
+        token_server = matches.group(2)
 
         # tell app that use this url to proceed with the authentication
-        self._redirect_auth_url(auth_server.strip('"'))
-
-        # go to the next step, request a token from the token server
-        self.__request_token()
-
-        return r
-
-    def __request_token(self):
-        self._thread_local.auth_step = self._AuthStep.GET_TOKEN
-        self._thread_local.attempts += 1
-
-        if self._thread_local.attempts < self.MAX_OAUTH_ATTEMPTS:
-            time.sleep(1)
-            self.http_session.get(self._thread_local.token_server, proxies=PROXIES)
-        else:
-            raise exceptions.TrinoAuthError("Exceeded max attempts while getting the token")
-
-    def __process_get_token(self, r):
-        token = None
-        if self._thread_local.attempts < self.MAX_OAUTH_ATTEMPTS:
-            response = r
-            if response.status_code == 200:
-                token_response = json.loads(response.text)
-                token = token_response.get('token')
-                if token:
-                    self._thread_local.token = token
-                    self._thread_local.attempts = 0
-                    self._thread_local.auth_step = self._AuthStep.GET_REDIRECT_SERVER
-                    return r
-                error = token_response.get('error')
-                if error:
-                    raise exceptions.TrinoAuthError(f"Error while getting the token : {error}")
+        self._redirect_auth_url(redirect_server)
+
+        # Consume content and release the original connection
+        # to allow our new request to reuse the same one.
+        response.content
```

Comment:
Is consuming the content required? The docs don't mention it explicitly.

Reply:
Right... that's a good point. I wasn't sure about it either but on the other hand [`HTTPDigestAuth`](https://github.com/psf/requests/blob/b0e025ade7ed30ed53ab61f542779af7e024932e/requests/auth.py#L261) handles reusing connections this way and this implementation is somewhat influenced by it.